admin/controllers/service-manager.js

```diff
@@ -226,92 +207,39 @@ const controller = {
    * @param next
    * @returns {Promise.<void>}
    */
-  saveCheckWindows: async (req, res, next) => {
-    let actionName = 'Create'
-    let urlActionName = 'add'
-    let checkWindow
-    let validationError = await checkWindowValidator.validate(req)
-    let currentYear = moment.utc(moment.now()).format('YYYY')
-    let flashMessage = req.body[ 'checkWindowName' ] + ' has been created'
-    let adminIsDisabled = req.body.adminIsDisabled
-    let checkStartIsDisabled = req.body.checkStartIsDisabled
-
-    if (req.body.checkWindowId !== '') {
-      actionName = 'Edit'
-      urlActionName = 'edit'
-      flashMessage = 'Changes have been saved'
-    }
+  saveCheckWindow: async (req, res, next) => {
+    let requestData = req.body
+    const getValidationResult = req.getValidationResult
+    const checkBody = req.checkBody
+    const validationError = await checkWindowValidator.validate(requestData, checkBody, getValidationResult)
+    const actionName = requestData.checkWindowId ? 'Edit' : 'Add'
+    const flashMessage = requestData.checkWindowId !== ''
+      ? 'Changes have been saved' : requestData['checkWindowName'] + ' has been created'
+    let adminIsDisabled = toBool(requestData.adminIsDisabled)
+    let checkStartIsDisabled = toBool(requestData.checkStartIsDisabled)
 
     if (validationError.hasError()) {
       res.locals.pageTitle = actionName + ' check window'
       req.breadcrumbs(res.locals.pageTitle)
-
-      if (!req.body[ 'adminStartDay' ] && !req.body[ 'adminStartMonth' ] && !req.body[ 'adminStartYear' ] && req.body[ 'existingAdminStartDate' ] && req.body[ 'adminIsDisabled' ] === '1') {
-        req.body.adminStartDay = moment(req.body[ 'existingAdminStartDate' ]).format('D')
-        req.body.adminStartMonth = moment(req.body[ 'existingAdminStartDate' ]).format('MM')
-        req.body.adminStartYear = moment(req.body[ 'existingAdminStartDate' ]).format('YYYY')
-      }
-
-      if (!req.body[ 'checkStartDay' ] && !req.body[ 'checkStartMonth' ] && !req.body[ 'checkStartYear' ] && req.body[ 'existingCheckStartDate' ] && req.body[ 'checkStartIsDisabled' ] === '1') {
-        req.body.checkStartDay = moment(req.body[ 'existingCheckStartDate' ]).format('D')
-        req.body.checkStartMonth = moment(req.body[ 'existingCheckStartDate' ]).format('MM')
-        req.body.checkStartYear = moment(req.body[ 'existingCheckStartDate' ]).format('YYYY')
-      }
-
+      requestData = checkWindowService.formatUnsavedData(requestData)
```

Comment:
This seems odd. 

Either a) this modifies the data which is passed by reference (not great) or
b) it returns a new data structure that is formatted (much better)

If b) then shouldn't it be assigned to a new variable?

Reply:
@jon-shipley just pushed another commit for this one